### PR TITLE
feat: upgrade chartjs from 2.x to 4.x, refactor chart download locally and remove quickchart dependency 

### LINKED ISSUE #1202: Refactor: move chart generation and download functionality from Quickcharts to MAPS GUI tool
## Summary

Refactor: move chart generation and download functionality from Quickcharts to MAPS GUI tool

## Description

- Quickcharts (https://github.com/typpo/quickchart) relies on https://github.com/SeanSobey/ChartjsNodeCanvas to render the charts. This repo currently only supports chartjs 3.x, whereas we need to use 4.x and beyond. In order to upgrade the MAPS GUI to chartjs 4.x we also need to upgrade Quickcharts to support 4.x and also node-canvas to 4.x, which means putting more resources into supporting forked versions of repos.
- Plan is to recreate the same functionality provided by Quickcharts, locally within the MAPS GUI.
- Charts are converted to base64 image locally and then placed within a styled canvas which includes the branding, QR code etc.
- This image will be then available for download as PNG and PDF

## Acceptance criteria

- [x] formatted image of the selected chart can be downloaded directly from the MAPS GUI via existing 'Download' buttons
- [x] user can download the image as PNG
- [x] user can download the image as PDF
- [x] Paulius Purple branding is applied
- [x] QR code generates link to quick maps URL
- [x] No external calls to Quickcharts endpoint remaining, all locally generated
- [x] Works in Chrome and Firefox

### PR DESCRIPTION
**Pull request by jlczapski**
## Summary
feat: upgrade chartjs from 2.x to 4.x, refactor chart download locally and remove quickchart dependency

### Upgrading chartjs from 2.x to 4.x

- see https://github.com/micronutrientsupport/micronutrient-support-tool/issues/1200 for full list of changes

### Refactor chart download locally and remove quickchart dependency

- Resolves #1202 
- No longer using custom Quickcharts functionality to download png/pdf.
- From Chart.js, we can get a Base64 image string, then use custom logic to draw canvas & add watermarks/QR code.
- Implemented `jsPDF` library to easily convert from img to PDF.
- User experience mirrors the previous version, just without calling external services.

### chart upgrade 2.x to 4.x

- [x] refer to table for locations of charts that have been updates and use develop branch version as means to compare current working vs new version
- [x] comfirm each chart renders data correctly
- [x] displays labels as expected
- [x] displays scales as expected
- [x] updates data when user changes variables e.g. mn
- [x] multiple charts within same card display as expected

### download charts 

- [x] charts download as pdf
- [x] charts download as png
- [x] charts display all the same branding as currently
- [x] charts download with correct titles and legend info

### CODE REVIEW

File: src/app/pages/costEffectiveness/interventionReview/utilities/reusableCostGraph/reusableCostGraph.component.ts [hunk 1]
```diff
@@ -48,92 +48,94 @@ export class ReusableCostGraphComponent implements OnInit {
...
-                    : getValueAtIndexOrDefault(ds.borderWidth, i, arcOpts.borderWidth);
+        maintainAspectRatio: false,
+        plugins: {
+          legend: {
+            display: true,
+            position: 'right',
+            onClick: () => {
+              return null;
+            },
+            // labels: { // TODO: chart fix labels
```
Comment (Fuhji) [inline src/app/pages/costEffectiveness/interventionReview/utilities/reusableCostGraph/reusableCostGraph.component.ts]: labels are still to be fixed on the pie charts

File: src/app/pages/costEffectiveness/interventionReview/utilities/reusableCostGraph/reusableCostGraph.component.ts [hunk 2]
```diff
@@ -48,92 +48,94 @@ export class ReusableCostGraphComponent implements OnInit {
...
-              const dataset_value = data.datasets[item.datasetIndex].data[item.index];
-              return (+dataset_value).toLocaleString('en-US', {
-                minimumFractionDigits: 2,
-                maximumFractionDigits: 2,
-                style: 'currency',
-                currency: 'USD',
-              });
-            },
+          tooltip: {
+            // callbacks: {
```
Comment (Fuhji) [inline src/app/pages/costEffectiveness/interventionReview/utilities/reusableCostGraph/reusableCostGraph.component.ts]: tooltips to be fixed

**Comment by jlczapski** [comment]
Issues to fix: 

- [x] Treemap download greyed out
- [x] Summary/detailed view in intervention cost summary, download greyed out
- [x] Intervention recurring costs, charts commented out?
- [x] Show threshold value on histogram chart

**Comment by bgsandan** [comment]
Download appears to be greyed out for top20 food treemap also:

image

**Comment by bgsandan** [comment]
Custom callout for the value of the threshold missing from the histogram (for household level i.e. Malawi)

image

For convenience, the threshold could be added to the label of the line (i.e. "Threshold for inadequacy (814)" If that is easier

**Comment by DanielLyleWilliamWarren** [comment]
LGTM, tested all graph views with Malawi Ca & Ethiopia Fe and downloaded all options. My only comment is the graph below doesn't take up all the space like it does in the current iteration of histogram graph: 
image

Other than that it all LGTM!

**Review by DanielLyleWilliamWarren — APPROVED** [review]
See comment but all LGTM!

**Pull request merged**